106_A2_M1/Model/Singleton.cs

```diff
@@ -95,15 +117,271 @@ public async Task<int> LoginAsync(string email, string password, string authToke
                 return 0;
             }
         }
+        private async Task<UserDB> GetUserInformationAsync()
+        {
+            try
+            {
+                // Make a request to your backend API to retrieve user information
+                HttpResponseMessage userResponse = await this._client.GetAsync("https://cse106-backend.d3rpp.dev/api/user");
+
+                if (userResponse.IsSuccessStatusCode)
+                {
+                    // Deserialize the response content to obtain the user information
+                    string content = await userResponse.Content.ReadAsStringAsync();
+                    var userData = JsonConvert.DeserializeObject<UserDataFromBackend>(content);
+
+                    // Map the backend data to UserDB class
+                    UserDB userDB = new UserDB
+                    {
+                        email = userData.email,
+                        first_name = userData.given_name,
+                        last_name = userData.family_name,
+                        dob = userData.dob_ts,
+                        id = userData.id,
+                        qr_status = userData.qrcode_status ?? 0, // Handle possible null value
+                        issue_ct = userData.issue_count,
+                        test_ct = userData.test_count,
+                        vaccine_status = userData.vaccine_status
+                    };
+
+                    return userDB;
+                }
+                else
+                {
+                    Console.WriteLine($"Error fetching user information: {userResponse.StatusCode} - {userResponse.ReasonPhrase}");
+                    return null;
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"An error occurred while fetching user information: {ex.Message}");
+                return null;
+            }
+        }
+
+        // Class to represent the data received from the backend
+        public class UserDataFromBackend
+        {
+            public string token { get; set; }
+            public string id { get; set; }
+            public string email { get; set; }
+            public string family_name { get; set; }
+            public string given_name { get; set; }
+            public string dob_ts { get; set; }
+            public int? qrcode_status { get; set; }
+            public int issue_count { get; set; }
+            public int test_count { get; set; }
+            public int vaccine_status { get; set; }
+        }
+
+        public async Task<int> CreateAccountAsync(string email, string password, string firstName, string lastName, int dob, int nhiNumber)
+        {
+            try
+            {
+                // Create a string containing the user data
+                UserData userData = new UserData
```

Comment:
I'll need to add a schema in the `Interfaces` folder for this one as well as this will not send correctly (JSON is case sensitive, hense the existence of the Interfaces)